File: backend/src/main/java/com/buildtrack/ai/service/MaterialService.java
package com.buildtrack.ai.service;

import com.buildtrack.ai.entity.Material;
import com.buildtrack.ai.entity.MaterialRequest;
import com.buildtrack.ai.entity.MaterialTransaction;

import java.util.List;

public interface MaterialService {
    List<Material> getVisibleMaterials(Long projectId);
    Material create(Material material);
    MaterialTransaction receive(Long materialId, MaterialTransaction transaction);
    MaterialTransaction issue(Long materialId, MaterialTransaction transaction);
    List<MaterialTransaction> history(Long materialId);

    MaterialRequest createRequest(MaterialRequest request);
    List<MaterialRequest> getRequests(Long projectId);
    MaterialRequest issueRequest(Long requestId);
    MaterialRequest workerReceiveRequest(Long requestId);
    MaterialRequest confirmRequest(Long requestId);
    void delete(Long id);
}

File: frontend/src/services/materialService.js
import api from './api';

const materialService = {
  async list(projectId) {
    const { data } = await api.get('/materials', {
      params: projectId ? { projectId } : undefined,
    });
    return data?.data || [];
  },
  async create(payload) {
    const { data } = await api.post('/materials', payload);
    return data?.data;
  },
  async receive(id, payload) {
    const { data } = await api.post(`/materials/${id}/receive`, payload);
    return data?.data;
  },
  async issue(id, payload) {
    const { data } = await api.post(`/materials/${id}/issue`, payload);
    return data?.data;
  },
  async history(id) {
    const { data } = await api.get(`/materials/${id}/history`);
    return data?.data || [];
  },
  async listRequests(projectId) {
    const { data } = await api.get('/materials/requests', {
      params: projectId ? { projectId } : undefined,
    });
    return data?.data || [];
  },
  async createRequest(payload) {
    const { data } = await api.post('/materials/requests', payload);
    return data?.data;
  },
  async issueRequest(id) {
    const { data } = await api.post(`/materials/requests/${id}/issue`);
    return data?.data;
  },
  async workerReceiveRequest(id) {
    const { data } = await api.post(`/materials/requests/${id}/worker-receive`);
    return data?.data;
  },
  async confirmRequest(id) {
    const { data } = await api.post(`/materials/requests/${id}/confirm`);
    return data?.data;
  },
  async remove(id) {
    const { data } = await api.delete(`/materials/${id}`);
    return data?.data;
  },
};

export default materialService;
